list/list-test.js

```diff
@@ -1311,3 +1312,98 @@ canTestHelpers.devOnlyTest("can.getName symbol behavior", function(assert) {
 		"should use custom list name when provided"
 	);
 });
+
+QUnit.test("length event should include previous value", function(assert) {
+	var done = assert.async();
+	var list = new DefineList([]);
+	var other = new DefineList(["a"]);
+
+	var changes = [];
+	list.on("length", function(_, current, previous) {
+		changes.push({ current: current, previous: previous });
+	});
+
+	list.push("x");
+	list.pop();
+	list.push("y", "z");
+	list.splice(2, 0, "x", "w");
+	list.splice(0, 1);
+	list.sort();
+	list.replace(other);
+
+	assert.expect(1);
+	setTimeout(function() {
```

Comment:
@justinbmeyer would you do this differently? I was wondering if I could use an observation to wrap the mutations and have a reliable way to know when it's done, done.